### DISCUSSION TITLE
Troubleshooting: Getting a KeyError when requiring a resource with a sensor in my repository

### DISCUSSION THREAD

**Discussion by prha** [dagster-io/dagster#14277] [Q&A] [answered]
I'm trying to use a resource in my sensor.

```python
@sensor(jobs=[my_job], required_resource_keys={'snowflake'})
def my_sensor(context):
    client = context.resources.snowflake.get_client()
    ...
```

But I'm getting an error `KeyError: 'snowflake'` when initializing my resource.

**Comment by prha** [comment, marked as answer]
Using `required_resource_keys` is supported only when the sensor is included in a `Definitions` class, which takes care of the resource binding for you.  If you are including your sensor in a `RepositoryDefinition` / `@repository`, you might hit this resource initialization error.

You can either switch from `@repository` => `Definitions` (preferred):
```python
defs = Definitions(
    jobs=[my_job],
    sensors=[my_sensor],
    resources={'snowflake': snowflake_resource},
)
```
Docs link: https://docs.dagster.io/concepts/partitions-schedules-sensors/sensors#using-resources-in-sensors


Or you can manually build the resource from within your sensor evaluation function:
```python
@sensor(jobs=[my_job])
def my_sensor(_):
   # initialize the resources using build_resources
    with build_resources(resources={"snowflake": snowflake_resource}) as resources:
        client = resources.snowflake.get_client()

```

Docs link: https://docs.dagster.io/_apidocs/resources#dagster.build_resources

**Reply by d4rk4dm** [reply]
is it possible to use here make_values_resource() when initializing resources?
